src/commands/init.js

```diff
@@ -248,22 +259,21 @@ function interactiveMenu(directory, list) {
     });
 }
 
-function checkFolder(force = false, directoryName = '') {
+function checkFolder(force = false, directory = '') {
     return new Promise((resolve) => {
-        const directoryPath = path.join(process.cwd(), directoryName);
+        const directoryPath = getAbsolutePath(directory || process.cwd());
```

Comment:
Good point!

I think the right thing here might be to update `getAbsolutePath` so that it does not return undefined when `filepath` is undefined but instead either `process.cwd()` or the directory that was provided to it. Seems like this is the expected result in the code but I will need to look a little closer to be 100% sure if we decide to go down this path.

When running `init` the `directory` will always be absolute when running in Roc. This means the cases 4 and 6 will never happen in the table above. Or rather the path will already be `/cwd/directory` when processing `init` from the cli. This means that some tests can be updated to reflect this.

I did these changes locally and the tests seems pass correctly, or all but one. `roc commands init should choose name over directory` does fail but I believe that it tests an incorrect/unexpected behaviour. For me the expected behaviour is the following:
```js
roc new name --directory directory // Should try init inside /cwd/directory/name 
```
---
The most important change this would bring is that all mentions of `process.cwd()` in the Roc __commands__ functions should be replaced with the new `directory` value that by default will be `process.cwd()` but can be configured by the user.